hnn_core/tests/test_network.py

```diff
@@ -44,6 +44,21 @@ def test_network():
         assert len(net.gid_ranges[dn]) == len(this_src_gids)
         assert len(net.external_drives[dn]['events']) == 1  # single trial!
 
+    # Test that a change in net.pos_dict influences the number of cells created
+    net_new = net.copy()
+    n_cells_orig = net_new.n_cells
+    new_cell_type = {'new_type': net_new.cell_types['L2_basket']}
+    net_new.pos_dict.update({'new_type': [(0, 0, 0)]})
+    net_new._update_gid_ranges()
+    net_new._update_cells()
+    # Without also updating net_new.cell_types, no new cells are created
```

Comment:
this test is a bit confusing. Ideally, an error should have been thrown in this scenario, no?

Reply:
I went back and forth on how to implement this test quite a bit. My original intention was to validate that `Network.cell_type` acts as a gate for creating cells. In theory, the user should be able to delete a cell type from the network without throwing an error; however, the API isn't completely setup yet to accommodate such a scenario. I simplified the test in https://github.com/jonescompneurolab/hnn-core/pull/321/commits/e1bfa153f4cec7c71185afb9ade1a72d091ced83 by removing this assertion statement.

Reply:
Does this work @jasmainak or would you prefer that we throw an error if/when the cell types in `pos_dict` and `cell_types` become incongruent?

Reply:
In a follow up PR, I think it would be great to put in checks to ensure user does not inconsistently update `pos_dict` / `cell_types`. One way is to make this attributes private so user is not allowed to touch them and have one function/method which can be used to update them consistently. Stuff like [this](https://github.com/rythorpe/hnn-core/blob/e9c79bf5cb3f9047ec463f00e9d9986965034121/hnn_core/network.py#L526) scares me ...